### PR TITLE
chore: expose compactor's hot compaction hours thresholds as cfg

### LINKED ISSUE #6059: Add config param for when to do hot compaction on a partition
See also #6038

Currently the number of hours before doing hot compactions is hardcoded. Make it configurable.

### PR DESCRIPTION
**Pull request by lukebond**
Closes #6059 

In #6044 Nga made it possible to configure how old a partition needs to be in order to be considered cold. This does the same for hot partitions.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: clap_blocks/src/compactor.rs [hunk 1]
```diff
@@ -176,6 +176,24 @@ macro_rules! gen_compactor_config {
...
+            pub hot_compaction_hours_threshold_1: u64,
+
+            /// When querying for partitions with data for hot compaction, how many hours to look
+            /// back for a second pass if we found nothing in the first pass.
+            #[clap(
+                long = "compaction-hot-partition-hours-threshold-2",
+                default_value = "24",
+                action
+            )]
+            pub hot_compaction_hours_threshold_2: u64,
```
Comment (NGA-TRAN) [inline clap_blocks/src/compactor.rs:196]: You want to add env = "INFLUXDB_IOX_COMPACTION..." for these 3 config param so we can add them in k8s_idpe

File: compactor/src/compact.rs [hunk 2]
```diff
@@ -258,7 +258,8 @@ impl Compactor {
     /// Return a list of partitions that:
     ///
     /// - Have not received any writes in 8 hours (determined by all level 0 and level 1 parquet
-    ///   files having a created_at time older than 8 hours ago)
+    ///   files having a created_at time older than 8 hours ago). Note that 8 is the default but
+    ///   it's configurable
```
Comment (NGA-TRAN) [inline compactor/src/compact.rs:262]: ❤️

File: compactor/src/compact.rs [hunk 3]
```diff
@@ -779,7 +785,7 @@ pub mod tests {
         // Create a compactor
         let time_provider = Arc::new(SystemProvider::new());
         let mut config = make_compactor_config();
-        // 8 hours to be cold
+        // 8 hours to be cold (this is the default)
```
Comment (NGA-TRAN) [inline compactor/src/compact.rs:788]: For tests, I created a config with time to be cold to 10 minutes. See here https://github.com/influxdata/influxdb_iox/blob/654ed98d1f5bb77243417e1e5de0738d5bd44d98/compactor/src/compact.rs#L692

In this test, I make it 8 hours so all behaviors in the tests are the same as it was with previous default 8 hours. I think it is appropriate not to mention the default here. We just test with 8 hours
```suggestion
        // 8 hours to be cold
```

File: compactor/src/compact.rs [hunk 4]
```diff
@@ -848,7 +854,7 @@ pub mod tests {
         assert!(candidates.is_empty());
 
         // --------------------------------------
-        // Case 3: no new recent writes (within the last 8 hours) --> return that partition
+        // Case 3: no new recent writes (within the last 8 hours, which is the default) --> return that partition
```
Comment (NGA-TRAN) [inline compactor/src/compact.rs:857]:
```suggestion
        // Case 3: no new recent writes (within the last 8 hours) --> return that partition
```

Same here

File: compactor/src/hot.rs [hunk 5]
```diff
@@ -84,7 +85,11 @@ pub(crate) async fn hot_partitions_to_compact(
 
     // Get the most recent highest ingested throughput partitions within the last 4 hours. If not,
     // increase to 24 hours.
-    let query_times = query_times(compactor.time_provider());
+    let query_times = query_times(
+        compactor.time_provider(),
+        compactor.config.hot_compaction_hours_threshold_1,
+        compactor.config.hot_compaction_hours_threshold_2,
```
Comment (NGA-TRAN) [inline compactor/src/hot.rs:91]: 🎉

**Review by NGA-TRAN — COMMENTED** [review]
I think if you add the 2 env for the new params it will be good. And minor comment changes

File: clap_blocks/src/compactor.rs (hunk 1 above)
Comment (lukebond) [reply clap_blocks/src/compactor.rs:196]: i thought it was done automatically but think i'm confusing it with our golang codebase

**Review by NGA-TRAN — APPROVED** [review]
Thanks @lukebond . If we add these 2 env params into idpe, we can adjust them now. I guess so far there are no demands to change them. We may just keep the settings 4 and 24 for now

**Pull request merged**